src/main/java/org/jabref/logic/importer/fetcher/ComplexSearchQuery.java

```diff
@@ -28,16 +31,32 @@ private ComplexSearchQuery(String defaultField, List<String> authors, List<Strin
         this.singleYear = singleYear;
     }
 
-    public Optional<String> getDefaultField() {
-        return Optional.ofNullable(defaultField);
+    public static ComplexSearchQuery fromTerms(Collection<Term> terms) {
+        ComplexSearchQueryBuilder builder = ComplexSearchQuery.builder();
+        terms.forEach(term -> {
+            String termText = term.text();
+            switch (term.field().toLowerCase()) {
+                case "author" -> builder.author(termText);
+                case "title" -> builder.titlePhrase(termText);
+                case "journal" -> builder.journal(termText);
+                case "year" -> builder.singleYear(Integer.valueOf(termText));
+                case "year-range" -> builder.fromYearAndToYear(Integer.valueOf(termText.split("-")[0]), Integer.valueOf(termText.split("-")[1]));
```

Comment:
`NumberFormatException` should be catched here.

See https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#valueOf(java.lang.String). I would use `0` and `9999` as default valuesto keep the search going.